optuna_dashboard/static/components/GraphContour.tsx

```diff
@@ -0,0 +1,280 @@
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+          <FormControl component="fieldset" className={classes.formControl}>
+            <InputLabel id="parameter2">Y Axis Parameter</InputLabel>
+            <Select value={yAxis} onChange={handleYAxisChange}>
+              {paramnames.map((x) => (
+                <MenuItem value={x} key={x}>
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </Grid>
+      </Grid>
+      <Grid item xs={6}>
+        <div id={plotDomId} />
+      </Grid>
+    </Grid>
+  )
+}
+
+const plotContour = (
+  study: StudyDetail,
+  objectiveId: number,
+  xAxis: string,
+  yAxis: string
+) => {
+  if (document.getElementById(plotDomId) === null) {
+    return
+  }
+
+  const layout: Partial<plotly.Layout> = {
+    title: "Contour",
+    margin: {
+      l: 50,
+      r: 50,
+    },
+  }
+
+  const trials: Trial[] = study !== null ? study.trials : []
+
+  if (trials.length === 0) {
+    plotly.react(plotDomId, [], layout)
+  }
+
+  const filteredTrials = trials.filter(
+    (t) => t.state === "Complete" || t.state === "Pruned"
+  )
+
+  let paramNames = new Set<string>(trials[0].params.map((p) => p.name))
+  filteredTrials.forEach((t) => {
+    paramNames = new Set<string>(
+      t.params.filter((p) => paramNames.has(p.name)).map((p) => p.name)
+    )
+  })
+
+  if (paramNames.size === 0 || paramNames.size === 1) {
+    plotly.react(plotDomId, [])
+    return
+  }
+  if (xAxis === yAxis) {
+    plotly.react(plotDomId, [], {
+      xaxis: {
+        gridcolor: "#f2f5fa",
+        gridwidth: 1,
+        zerolinecolor: "#f2f5fa",
+        zerolinewidth: 1.5,
+      },
+      yaxis: {
+        gridcolor: "#f2f5fa",
+        gridwidth: 1,
+        zerolinecolor: "#f2f5fa",
+        zerolinewidth: 1.5,
+      },
+      plot_bgcolor: "#E5ecf6",
+    })
+    return
+  }
+
+  const objectiveValues: number[] = filteredTrials.map(
+    (t) => t.values![objectiveId]
+  )
+
+  const paramValues: { [key: number]: string[] } = []
+  const paramIndices: Array<string> = []
+
+  if (paramNames.size >= 2) {
+    let i = 0
+    paramNames.forEach((paramName) => {
+      const valueStrings = filteredTrials.map((t) => {
+        const param = t.params.find((p) => p.name == paramName)
+        return param!.value
+      })
+      paramValues[i] = valueStrings
+      i++
+    })
+
+    let k = 0
+    paramNames.forEach((paramName) => {
+      paramIndices[k] = paramName
+      k++
+    })
+    const paramCategorical = { ...paramValues }
+    const xIndex = paramIndices.indexOf(xAxis)
+    const yIndex = paramIndices.indexOf(yAxis)
+
+    const x_indice = paramCategorical[xIndex].sort((a, b) => (a > b ? 1 : -1))
+    const y_indice = paramCategorical[yIndex].sort((a, b) => (a > b ? 1 : -1))
+
+    const x_indices: string[] = []
+    const y_indices: string[] = []
```

Comment:
Could you avoid using snake_case_with_underscores? Other variables are named in camelCase styles.

Reply:
Yes, sure! I'll adhere to that convention.